File: apps/api/AssetManagement.Api/DTOs/AuthDtos.cs
namespace AssetManagement.Api.DTOs;

public record LoginRequest(string Username, string Password);

public record LoginResponse(string Token, UserProfileResponse User);

public record UserProfileResponse(
    Guid Id,
    string Username,
    string DisplayName,
    string Email,
    List<string> Roles,
    string? ThemePreference = null);

File: apps/api-kt/src/main/kotlin/com/assetmanagement/api/dto/AuthDtos.kt
package com.assetmanagement.api.dto

import java.util.*

data class UserProfileResponse(
    val id: UUID,
    val username: String,
    val displayName: String,
    val email: String,
    val roles: List<String>,
    val themePreference: String? = null,
    val authProvider: String = "LOCAL"
)
